core/src/dbs/store.rs

```diff
@@ -13,11 +15,56 @@ impl MemoryCollector {
 		self.0.push(val);
 	}
 
-	pub(super) fn sort(&mut self, ordering: &Ordering) {
+	/// This function determines the sorting strategy based on the size of the collection.
+	/// If the collection contains fewer than 1000 elements, it uses `small_sort`.
+	/// Otherwise, it uses `large_sort` which is asynchronous.
+	///
+	#[cfg(not(target_arch = "wasm32"))]
+	pub(super) async fn sort(&mut self, ordering: &Ordering) {
+		if self.0.len() < 1000 {
+			self.small_sort(ordering);
+		} else {
+			self.large_sort(ordering).await;
+		}
+	}
+
+	#[cfg(not(target_arch = "wasm32"))]
+	/// Asynchronously sorts a large vector based on the given ordering.
+	///
+	/// The function performs the sorting operation in a blocking
+	/// manner to avoid occupying the async runtime,
+	/// and then awaits the completion of the sorting.
+	///
+	/// - For vectors with a length of 10,000 or more, the sorting is performed using `par_sort_unstable_by`
+	///   from the Rayon library for better performance through parallelism.
+	/// - For smaller vectors, standard `sort_by` is used.
+	///
+	async fn large_sort(&mut self, ordering: &Ordering) {
+		let mut vec = mem::take(&mut self.0);
+		let ordering = ordering.clone();
+		let vec = spawn_blocking(move || {
+			match ordering {
+				Ordering::Random => vec.shuffle(&mut rand::thread_rng()),
+				Ordering::Order(orders) => {
+					if vec.len() >= 10000 {
+						vec.par_sort_unstable_by(|a, b| orders.compare(a, b));
+					} else {
+						vec.sort_unstable_by(|a, b| orders.compare(a, b));
+					}
+				}
+			};
+			vec
+		})
+		.await
+		.unwrap();
+		self.0 = vec;
+	}
+
+	pub(super) fn small_sort(&mut self, ordering: &Ordering) {
 		match ordering {
 			Ordering::Random => self.0.shuffle(&mut rand::thread_rng()),
 			Ordering::Order(orders) => {
-				self.0.sort_by(|a, b| orders.compare(a, b));
+				self.0.sort_unstable_by(|a, b| orders.compare(a, b));
```

Comment:
@tobiemh If sorting can be unstable when using Rayon, should we switch to `sort_unstable_by` for the normal sort as well?